controllers/storageoscluster/node-manager_operand.go

```diff
@@ -187,16 +182,28 @@ func (c *NodeManagerOperand) PostReady(ctx context.Context, obj client.Object) e
 				log.Info("resource has deleted")
 
 				c.watchLock <- true
-				defer func() {
-					<-c.watchLock
-				}()
 				c.watcher = nil
+				<-c.watchLock
+
+				close(closeWatchWaiter)
 
 				return nil
 			}
 		}
 	})
 
+	go func() {
+		select {
+		case <-closeWatchWaiter:
+			return
+		case err := <-watchErrChan:
+			log.Error(err, "node watcher has some permanent issues")
```

Comment:
"has some permanent issues" doesn't add any value to the msg
Also, the only err I see being passed to the chan is when `waitPeriod >= maxWaitPeriod`. If that's the case, add a meaningful log msg at that point and don't log anything here

Reply:
not agree, at this point we don't know what is the reason. More details found in err. We can't relay on the implementation what we call. At this point we know, something went wrong. error will contain the reason.

Reply:
I had to look at the code locally to understand it better. I agree to not remove the logging but I stand with how I feel about "has some permanent issues" needing to be written in another way or removed

Reply:
@nolancon @Mojachieee could you give your thoughts on this?
I am too picky with how log msgs should be

Reply:
`config map watcher failed` seem good to me

Reply:
isn't it a Daemonset watcher? 
Either way, "..... encountered error" gets my vote